src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java

```diff
@@ -147,22 +159,39 @@ public DrivebaseSubsystem() {
 	 * @param rotation
 	 * @param fieldOriented
 	 */
-	public void drive(double forward, double strafe, Rotation2d rotation, boolean fieldOriented) {
-		SwerveModuleState[] moduleStates = getModuleStates(new ChassisSpeeds(0, 0, 0));
+	public void drive(
+			double forward,
+			double strafe,
+			Rotation2d rotation,
+			boolean fieldOriented,
+			boolean autoBalance) {
+		// Auto balancing will only be used in autonomous
+		if (autoBalance) {
+			forward -= balanceController.update((double) gyroscope.getRoll());
+		}
+
+		ChassisSpeeds chassisSpeeds = new ChassisSpeeds(0, 0, 0);
+
 		if (fieldOriented) {
-			moduleStates =
-					getModuleStates(
-							ChassisSpeeds.fromFieldRelativeSpeeds(
-									forward, -strafe, rotation.getRadians(), getGyroRotation2d()));
+			chassisSpeeds =
+					ChassisSpeeds.fromFieldRelativeSpeeds(
+							forward, -strafe, rotation.getRadians(), getGyroRotation2d());
 		} else {
-			moduleStates =
-					getModuleStates(new ChassisSpeeds(forward, -strafe, rotation.getRadians() * 100));
+			chassisSpeeds = new ChassisSpeeds(forward, -strafe, rotation.getRadians());
 		}
-		drive(moduleStates);
+		drive(chassisSpeeds);
 	}
 
 	public void drive(ChassisSpeeds chassisSpeeds) {
 		SwerveModuleState[] moduleStates = getModuleStates(chassisSpeeds);
+		if (Math.abs(chassisSpeeds.vxMetersPerSecond) <= 0.01
+				&& Math.abs(chassisSpeeds.vyMetersPerSecond) <= 0.01
+				&& Math.abs(chassisSpeeds.omegaRadiansPerSecond) <= 0.01) {
+			moduleStates[0] = new SwerveModuleState(0, Rotation2d.fromDegrees(45));
```

Comment:
At some point it should handle the different wheel placements of bonk and the new robot.

Reply:
Yes, these sorts of issues are being (slowly and painfully) dealt with at https://github.com/jbko6/ChargedUp2023/tree/pidgeon